satpy/readers/eum_l2_bufr.py

```diff
@@ -84,54 +94,95 @@ class SeviriL2BufrFileHandler(BaseFileHandler):
 
     def __init__(self, filename, filename_info, filetype_info, with_area_definition=False,
                  rectification_longitude='default', **kwargs):
-        """Initialise the file handler for SEVIRI L2 BUFR data."""
-        super(SeviriL2BufrFileHandler, self).__init__(filename,
-                                                      filename_info,
-                                                      filetype_info)
+        """Initialise the file handler for EUMETSAT SEVIRI and FCI L2 BUFR data."""
+        super(EumetsatL2BufrFileHandler, self).__init__(filename,
+                                                        filename_info,
+                                                        filetype_info)
 
         if ('server' in filename_info):
             # EUMETSAT Offline Bufr product
-            self.mpef_header = self._read_mpef_header()
+            self.bufr_header = self._read_mpef_header()
         else:
             # Product was retrieved from the EUMETSAT Data Center
-            timeStr = self.get_attribute('typicalDate')+self.get_attribute('typicalTime')
-            buf_start_time = datetime.strptime(timeStr, "%Y%m%d%H%M%S")
-            sc_id = self.get_attribute('satelliteIdentifier')
-            self.mpef_header = {}
-            self.mpef_header['NominalTime'] = buf_start_time
-            self.mpef_header['SpacecraftName'] = data_center_dict[sc_id]['name']
-            self.mpef_header['RectificationLongitude'] = data_center_dict[sc_id]['ssp']
+            # get all attributes in one call
+            attr = self.get_attributes(['typicalDate', 'typicalTime'])
+            timeStr = attr['typicalDate']+attr['typicalTime']
+
+            darr = self.get_array('satelliteIdentifier')
+            arr = da.where(darr != BUFR_FILL_VALUE, darr, darr).compute()
+            if (arr.ndim > 0):
+                sc_id = int(arr[0])
+            else:
+                sc_id = int(arr)
```

Comment:
The proposed method change (I tried it before) doesn't work because with the change to the BUFR library outputs with the environment variable, requesting the satelliteIdentifier key returns an array (while typicalDate and typicalTime are still scalars). So that's why satelliteIdentifier and segmentSizeAtNadirInXDirection are trated separately.
Do you know another way to get a value from an array without having to apply .compute() first ? I didn't find any so why would satpy be picky about it?